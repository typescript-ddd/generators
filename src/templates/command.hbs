import { Command } from "@typescript-ddd/core";

/**
 * Represents a command for {{actionVerb actionName}} {{noun entityName}}. 
 */
export class {{action actionName}}{{pascal entityName}}Command implements Command {

  /**
   * Initializes a new instance of the {{actionName}} {{natural entityName}} command.
   * @param {{lbrace}}{{pascal entityName}}Id{{rbrace}} id - the identifier of the {{natural entityName}}.
   {{#each properties}}
   * @param {{lbrace}}{{type}}{{rbrace}} {{name}} - the {{name}} of the {{natural ../entityName}}.
   {{/each}}
   */
  constructor(
    public readonly id: {{pascal entityName}}Id,
    {{#each properties}}
    public readonly {{name}}: {{type}},
    {{/each}}
  ) {
    // TODO: Add validation here
  }
}
