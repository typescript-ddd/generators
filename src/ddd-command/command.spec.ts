import { CommandTemplateData } from "./command.types";
import { getCommandTemplate } from "./command.generator";

describe("Command", () => {
  const template = getCommandTemplate();

  it("should render template", () => {
    // Arrange
    const data: CommandTemplateData = {
      entityName: "User",
      actionName: "create",
      properties: [
        { name: "name", type: "string" },
        { name: "email", type: "string" },
      ],
    };

    // Act
    const result = template(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
