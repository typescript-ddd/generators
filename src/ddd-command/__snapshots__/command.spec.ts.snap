// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command should render template 1`] = `
"import { Command } from "@typescript-ddd/core";

/**
 * Represents a command for creating a user. 
 */
export class CreateUserCommand implements Command {

  /**
   * Initializes a new instance of the create user command.
   * @param {UserId} id - the identifier of the user.
   * @param {string} name - the name of the user.
   * @param {string} email - the email of the user.
   */
  constructor(
    public readonly id: UserId,
    public readonly name: string,
    public readonly email: string,
  ) {
    // TODO: Add validation here
  }
}
"
`;
